pkg/loki/modules.go

```diff
@@ -211,6 +211,8 @@ func (t *Loki) initIngester() (_ services.Service, err error) {
 	logproto.RegisterIngesterServer(t.server.GRPC, t.ingester)
 	grpc_health_v1.RegisterHealthServer(t.server.GRPC, t.ingester)
 	t.server.HTTP.Path("/flush").Handler(http.HandlerFunc(t.ingester.FlushHandler))
+	// TODO(owen-d): should this use cortex style path (/ingester/shutdown), legacy style (/shutdown), or apir prefixed (/loki/api/v1/ingester/shutdown)?
+	t.server.HTTP.Methods("POST").Path("/ingester/shutdown").Handler(http.HandlerFunc(t.ingester.ShutdownHandler))
```

Comment:
Q: How is this used ?

Reply:
From the `ShutdownHandler` code:
```
// ShutdownHandler triggers the following set of operations in order:
//     * Change the state of ring to stop accepting writes.
//     * Flush all the chunks.
```

This is helpful when you manuallly want to scale down ingesters and don't want to leave unflushed data in orphaned WALs.

Reply:
From the `ShutdownHandler` code:
```
// ShutdownHandler triggers the following set of operations in order:
//     * Change the state of ring to stop accepting writes.
//     * Flush all the chunks.
```

This is helpful when you manuallly want to scale down ingesters and don't want to leave unflushed data in orphaned WALs.